staker/block_validator.go

```diff
@@ -866,7 +875,38 @@ validationsLoop:
 				nonBlockingTrigger(v.testingProgressMadeChan)
 			}
 			log.Trace("result validated", "count", v.validated(), "blockHash", v.lastValidGS.BlockHash)
-			continue
+		}
+	}
+}
+
+// return val:
+// *MessageIndex - pointer to bad entry if there is one (requires reorg)
+func (v *BlockValidator) sendValidations(ctx context.Context) (*arbutil.MessageIndex, error) {
+	v.reorgMutex.RLock()
+	defer v.reorgMutex.RUnlock()
+
+	wasmRoots := v.GetModuleRootsToValidate()
+	pos := v.lastValidationSent()
+	sendValidationUntil := v.validated() + arbutil.MessageIndex(v.config().ValidationSentLimit)
+	for pos <= sendValidationUntil {
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
+		v.reorgMutex.RUnlock()
+		v.reorgMutex.RLock()
```

Comment:
after getting the reorgMutex, you need to read lastValidationSent() into pos again, becaus e if there was a reorg it could have changed that pointer.